File: src/code-samples/langsmith/smithdb-migration/traces-list-runs-filter-after.go

// :snippet-start: traces-list-runs-filter-after-go
// :codegroup-tab: After
package main

import (
	"context"
	"time"

	"github.com/langchain-ai/langsmith-go"
)

func main() {
	ctx := context.Background()
	client := langsmith.NewClient()

	sessions, err := client.Sessions.List(ctx, langsmith.SessionListParams{
		Name:  langsmith.F("default"),
		Limit: langsmith.F(int64(1)),
	})
	if err != nil {
		panic(err.Error())
	}
	projectID := sessions.Items[0].ID
	traceID := "<trace-id>"
	// :remove-start:
	minStart, _ := time.Parse(time.RFC3339, "2026-07-01T00:00:00Z")
	maxStart, _ := time.Parse(time.RFC3339, "2026-07-31T23:59:59Z")
	iter := client.Traces.QueryAutoPaging(ctx, langsmith.TraceQueryParams{
		ProjectID:    langsmith.F(projectID),
		MinStartTime: langsmith.F(minStart),
		MaxStartTime: langsmith.F(maxStart),
	})
	if iter.Next() {
		traceID = iter.Current().RootRun.TraceID
	}
	// :remove-end:

	_, err = client.Traces.ListRuns(ctx, traceID, langsmith.TraceListRunsParams{
		ProjectID: langsmith.F(projectID),
		Filter:    langsmith.F(`eq(run_type, "llm")`),
		Selects: langsmith.F([]langsmith.TraceListRunsParamsSelect{
			langsmith.TraceListRunsParamsSelectName,
			langsmith.TraceListRunsParamsSelectStatus,
		}),
	})
	if err != nil {
		panic(err.Error())
	}
}
// :snippet-end:

File: src/code-samples/langsmith/smithdb-migration/traces-list-runs-filter-after.ts

// :snippet-start: traces-list-runs-filter-after-js
// :codegroup-tab: After
import { Client } from "langsmith";

const client = new Client();
const project = await client.readProject({ projectName: "default" });
let traceId = "<trace-id>";
// :remove-start:
for await (const t of client.traces.query({
  project_id: project.id,
  min_start_time: "2026-07-01T00:00:00Z",
  max_start_time: "2026-07-31T23:59:59Z",
})) {
  traceId = t.root_run!.trace_id!;
  break;
}
// :remove-end:
const response = await client.traces.listRuns(traceId, {
  project_id: project.id,
  filter: 'eq(run_type, "llm")',
  selects: ["NAME", "STATUS"],
});
const llmRuns = response.items ?? [];
// :snippet-end:
